compiler/semantic/sql.go

```diff
@@ -47,33 +39,19 @@ func (a *analyzer) semSelect(sel *ast.Select, seq dag.Seq) dag.Seq {
 			a.error(sel, errors.New("aggregate mixed with *-selector not yet supported"))
 			return append(seq, badOp())
 		}
-		groupby, ok := a.semGroupBy(sel.GroupBy, proj)
+		seq, ok = a.semGroupBy(sel.GroupBy, proj, seq)
 		if !ok {
-			seq = append(seq, badOp())
-		} else {
-			seq = append(seq, groupby)
-			if sel.Having != nil {
-				seq = append(seq, dag.NewFilter(a.semExpr(sel.Having)))
-			}
+			return seq
 		}
-	} else if sel.Args != nil {
 		if sel.Having != nil {
-			a.error(sel, errors.New("HAVING clause used without GROUP BY"))
-			seq = append(seq, badOp())
+			seq = append(seq, dag.NewFilter(a.semExpr(sel.Having)))
 		}
-		// GroupBy will do the cutting but if there's no GroupBy,
-		// then we need a cut for the select expressions.
-		// For SELECT *, cutter is nil.
-		selector, err := projectSelect(proj)
-		if err != nil {
-			a.error(sel, err)
-			seq = append(seq, badOp())
-		} else {
-			seq = append(seq, selector)
+	} else if sel.Selection.Args != nil {
+		if sel.Having != nil {
+			a.error(sel, errors.New("HAVING clause used without GROUP BY"))
```

Comment:
Shouldn't the error be on the having clause?
```suggestion
			a.error(sel.Having, errors.New("HAVING clause used without GROUP BY"))
```

Reply:
good catch!